enterprise/j2ee.core/src/org/netbeans/api/j2ee/core/Profile.java

```diff
@@ -140,134 +113,20 @@ public String toString() {
-        }
-        return profileDetails;
-    }
+        String valueMinusQuotes = value.replace("\"","");
 
-    private static boolean compareWebAndFull(@NonNull Profile profileToCompare, @NonNull Profile comparingVersion) {
-        boolean isThisFullProfile = isFullProfile(profileToCompare);
-        boolean isParamFullProfile = isFullProfile(comparingVersion);
-        if (isThisFullProfile && isParamFullProfile) {
-            // Both profiles are Java EE Full
-            return true;
-        }
-        if (!isThisFullProfile && !isParamFullProfile) {
-            // Both profiles are Java EE Web
-            return true;
-        }
-        if (isThisFullProfile && !isParamFullProfile) {
-            // profileToCompare is Java EE Full profile and comparingVersion is only Java EEWeb profile
-            return true;
-        }
-        return false;
-    }
-
-    private static boolean isFullProfile(@NonNull Profile profile) {
-        final String profileDetails = profile.toPropertiesString();
-        if (profileDetails.indexOf("-") == -1) {
-            return true;
-        }
-        return false;
-    }
-
-    /**
-     * Compares if the first given profile has equal or higher Java EE version
-     * in comparison to the second profile.
-     *
-     * Please be aware of the following rules:
-     * <br/><br/>
-     *
-     * 1) Each Java EE X version is considered as lower than Java EE X+1 version
-     * (this applies regardless on Web/Full specification and in reality it means
-     * that even Java EE 6 Full version is considered as lower than Java EE 7 Web)
-     * <br/><br/>
-     *
-     * 2) Each Java EE X Web version is considered as lower than Java EE X Full
-     * <br/>
-     *
-     * @param profileToCompare profile that we want to compare
-     * @param comparingVersion version which we are comparing with
-     * @return <code>true</code> if the profile version is equal or higher in
-     *         comparison with the second one, <code>false</code> otherwise
-     * @since 1.19
-     */
-    private static boolean isVersionEqualOrHigher(@NonNull Profile profileToCompare, @NonNull Profile comparingVersion) {
-        int comparisonResult = Profile.UI_COMPARATOR.compare(profileToCompare, comparingVersion);
-        if (comparisonResult == 0) {
-            // The same version for both
-            return true;
-        } else {
-            String profileToCompareVersion = getProfileVersion(profileToCompare);
-            String comparingProfileVersion = getProfileVersion(comparingVersion);
-            if (profileToCompareVersion.equals(comparingProfileVersion)) {
-                return compareWebAndFull(profileToCompare, comparingVersion);
-            } else {
-                // profileToCompare has lower version than comparingVersion
-                return comparisonResult <= 0;
+        for (Profile profile : Profile.values()) {
+            if (profile.toPropertiesString().equals(valueMinusQuotes)
```

Comment:
@asbachb what we could do is to make it a `"10.0"` in the profile, but change this line to:
```java
if (profile.toPropertiesString().equals(valueMinusQuotes) || profile.toPropertiesString().equals(valueMinusQuotes+".0"))
```
This wouldn't break backwards compatibility while also not breaking consistency. What do you think?

Reply:
nevermind, this wouldn't work due to `-web`

Reply:
lets keep it as is. At some point in future we can do the following:
 - remove the overridden `toString()`
 - return the enum `name()` in `toPropertiesString()`

This will get rid of the properties mess for new projects and simply use the enum name. The `fromPropertiesString()` method is already supporting it.

Reply:
> @asbachb what we could do is to make it a `"10.0"` in the profile, but change this line to:
> 
> ```java
> if (profile.toPropertiesString().equals(valueMinusQuotes) || profile.toPropertiesString().equals(valueMinusQuotes+".0"))
> ```
> 
> This wouldn't break backwards compatibility while also not breaking consistency. What do you think?

I'd say if there's this is not causing any issues somewhere else we should live with the current implementation. I don't see the benefit in changing it when this means to do major refactoring, think about possible backward compatibilities or introduce some quirks.